From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Irochi <me@irochi.moe>
Date: Thu, 15 Sep 2022 15:33:08 +0900
Subject: [PATCH] Rubia Rebrand


diff --git a/build.gradle.kts b/build.gradle.kts
index e950bce5047552bdd3f5664eb24ce290b0a06225..0e93b317331a8ef01b266900297c58984ef4905f 100644
--- a/build.gradle.kts
+++ b/build.gradle.kts
@@ -95,6 +95,8 @@ tasks.jar {
 }
 
 tasks.withType<Javadoc> {
+    (options as StandardJavadocDocletOptions).addStringOption("Xdoclint:none", "-quiet") // Rubia - silence Paper's bajillion javadoc warnings
+    (options as StandardJavadocDocletOptions).addStringOption("-add-modules", "jdk.incubator.vector") // Rubia
     val options = options as StandardJavadocDocletOptions
     options.overview = "src/main/javadoc/overview.html"
     options.use()
