From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Irochi <me@irochi.moe>
Date: Fri, 2 Sep 2022 16:49:46 +0900
Subject: [PATCH] Disable enfore secure profile config by default


diff --git a/src/main/java/net/minecraft/server/dedicated/DedicatedServerProperties.java b/src/main/java/net/minecraft/server/dedicated/DedicatedServerProperties.java
index c7e4330c93baff1f3027d7c75cf857b673d38970..0baf1a2a40579059dbf6ddcec3292aaa9c3f299e 100644
--- a/src/main/java/net/minecraft/server/dedicated/DedicatedServerProperties.java
+++ b/src/main/java/net/minecraft/server/dedicated/DedicatedServerProperties.java
@@ -154,7 +154,7 @@ public class DedicatedServerProperties extends Settings<DedicatedServerPropertie
         this.textFilteringConfig = this.get("text-filtering-config", "");
         this.playerIdleTimeout = this.getMutable("player-idle-timeout", 0);
         this.whiteList = this.getMutable("white-list", false);
-        this.enforceSecureProfile = this.get("enforce-secure-profile", true);
+        this.enforceSecureProfile = this.get("enforce-secure-profile", false); // Rubia
         String s = this.get("level-seed", "");
         boolean flag = this.get("generate-structures", true);
         long i = WorldOptions.parseSeed(s).orElse(WorldOptions.randomSeed());
