From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Irochi <me@irochi.moe>
Date: Tue, 25 Oct 2022 01:15:52 +0900
Subject: [PATCH] Better offline warning message when using Velocity


diff --git a/src/main/java/net/minecraft/server/dedicated/DedicatedServer.java b/src/main/java/net/minecraft/server/dedicated/DedicatedServer.java
index 1065d0b31406fc45cc54e44902bd6f170462e5ea..0f11355863e61a81b5e5e257657b62c12c4e8a48 100644
--- a/src/main/java/net/minecraft/server/dedicated/DedicatedServer.java
+++ b/src/main/java/net/minecraft/server/dedicated/DedicatedServer.java
@@ -284,6 +284,8 @@ public class DedicatedServer extends MinecraftServer implements ServerInterface
             if (org.spigotmc.SpigotConfig.bungee) {
                 DedicatedServer.LOGGER.warn("Whilst this makes it possible to use BungeeCord, unless access to your server is properly restricted, it also opens up the ability for hackers to connect with any username they choose.");
                 DedicatedServer.LOGGER.warn("Please see http://www.spigotmc.org/wiki/firewall-guide/ for further information.");
+            } else if (io.papermc.paper.configuration.GlobalConfiguration.get().proxies.velocity.enabled) {
+                DedicatedServer.LOGGER.warn("The server is configured to use Velocity. Make sure you have set correct forwarding secret key in both Paper and Velocity's configuration file."); // Rubia
             } else {
                 DedicatedServer.LOGGER.warn("While this makes the game possible to play without internet access, it also opens up the ability for hackers to connect with any username they choose.");
             }
