From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Irochi <me@irochi.moe>
Date: Tue, 25 Oct 2022 15:18:50 +0900
Subject: [PATCH] Add option to disable out-of-order-chat


diff --git a/src/main/java/moe/irochi/rubia/RubiaConfig.java b/src/main/java/moe/irochi/rubia/RubiaConfig.java
index 6fca938e7fb31c82f90c48ebb45385768f5f21c3..2433255f45ca520cd5a6e6f2be9ecaef94b9f186 100644
--- a/src/main/java/moe/irochi/rubia/RubiaConfig.java
+++ b/src/main/java/moe/irochi/rubia/RubiaConfig.java
@@ -132,4 +132,8 @@ public class RubiaConfig {
         return config.getStringList(key);
     }
 
+    public static boolean disableOutOfOrderChat;
+    private static void miscSettings() {
+        disableOutOfOrderChat = getBoolean("disable-out-of-order-chat", false);
+    }
 }
diff --git a/src/main/java/net/minecraft/server/network/ServerGamePacketListenerImpl.java b/src/main/java/net/minecraft/server/network/ServerGamePacketListenerImpl.java
index 0c2255b6e2fb7752f85b0f83d4f84732758bd14d..d7c0d61fc129b1b2c6c697f034757c18dc24fc09 100644
--- a/src/main/java/net/minecraft/server/network/ServerGamePacketListenerImpl.java
+++ b/src/main/java/net/minecraft/server/network/ServerGamePacketListenerImpl.java
@@ -2354,6 +2354,7 @@ public class ServerGamePacketListenerImpl implements ServerPlayerConnection, Tic
     }
 
     private boolean updateChatOrder(Instant timestamp) {
+        if (moe.irochi.rubia.RubiaConfig.disableOutOfOrderChat) return true; // Rubia
         Instant instant1;
 
         do {
