From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Irochi <me@irochi.moe>
Date: Fri, 11 Nov 2022 20:46:55 +0900
Subject: [PATCH] Allow any characters in username


diff --git a/src/main/java/moe/irochi/rubia/RubiaConfig.java b/src/main/java/moe/irochi/rubia/RubiaConfig.java
index 2e49a099cbff0b4da1f6b2a30c55a43e29380d5f..d644322e2f0088aa7eabb109ceeaed7afce855ad 100644
--- a/src/main/java/moe/irochi/rubia/RubiaConfig.java
+++ b/src/main/java/moe/irochi/rubia/RubiaConfig.java
@@ -133,8 +133,10 @@ public class RubiaConfig {
     }
 
     public static boolean offlineWarning;
+    public static boolean allowAnyUsername;
     private static void miscSettings() {
         offlineWarning = getBoolean("offline-warning", true);
+        allowAnyUsername = getBoolean("allow-any-username", false);
     }
 
 }
diff --git a/src/main/java/net/minecraft/server/network/ServerLoginPacketListenerImpl.java b/src/main/java/net/minecraft/server/network/ServerLoginPacketListenerImpl.java
index 3f7fab2f80ded9a7a720bd623f1ef028a4d0dca6..ea49cc169310a08d07b45dd26558874182941b1c 100644
--- a/src/main/java/net/minecraft/server/network/ServerLoginPacketListenerImpl.java
+++ b/src/main/java/net/minecraft/server/network/ServerLoginPacketListenerImpl.java
@@ -237,7 +237,7 @@ public class ServerLoginPacketListenerImpl implements ServerLoginPacketListener,
     @Override
     public void handleHello(ServerboundHelloPacket packet) {
         Validate.validState(this.state == ServerLoginPacketListenerImpl.State.HELLO, "Unexpected hello packet", new Object[0]);
-        Validate.validState(ServerLoginPacketListenerImpl.isValidUsername(packet.name()), "Invalid characters in username", new Object[0]);
+        if (!moe.irochi.rubia.RubiaConfig.allowAnyUsername) Validate.validState(ServerLoginPacketListenerImpl.isValidUsername(packet.name()), "Invalid characters in username", new Object[0]); // Rubia
         // Paper start - validate usernames
         if (io.papermc.paper.configuration.GlobalConfiguration.get().proxies.isProxyOnlineMode() && io.papermc.paper.configuration.GlobalConfiguration.get().unsupportedSettings.performUsernameValidation) {
             if (!this.iKnowThisMayNotBeTheBestIdeaButPleaseDisableUsernameValidation && !validateUsername(packet.name())) {
